src/main/java/org/opentripplanner/gtfs/mapping/StopMapper.java

```diff
@@ -43,59 +46,58 @@ private Stop doMap(org.onebusaway.gtfs.model.Stop gtfsStop) {
     }
 
     StopMappingWrapper base = new StopMappingWrapper(gtfsStop);
-
-    // Map single GTFS ZoneId to OTP TariffZone collection
-    Collection<FareZone> fareZones = getTariffZones(
-      gtfsStop.getZoneId(),
-      gtfsStop.getId().getAgencyId()
-    );
-
-    final I18NString name = translationHelper.getTranslation(
-      org.onebusaway.gtfs.model.Stop.class,
-      "name",
-      base.getId().getId(),
-      base.getName()
+    StopBuilder builder = Stop
+      .of(base.getId())
+      .withCode(base.getCode())
+      .withCoordinate(base.getCoordinate())
+      .withWheelchairAccessibility(base.getWheelchairAccessibility())
+      .withLevel(base.getLevel())
+      .withPlatformCode(gtfsStop.getPlatformCode())
+      .withVehicleType(TransitModeMapper.mapMode(gtfsStop.getVehicleType()));
+
+    builder.withName(
+      translationHelper.getTranslation(
+        org.onebusaway.gtfs.model.Stop.class,
+        "name",
+        base.getId().getId(),
+        base.getName()
+      )
     );
 
-    final I18NString desc = translationHelper.getTranslation(
-      org.onebusaway.gtfs.model.Stop.class,
-      "desc",
-      base.getId().getId(),
-      base.getDescription()
+    builder.withDescription(
+      translationHelper.getTranslation(
+        org.onebusaway.gtfs.model.Stop.class,
+        "desc",
+        base.getId().getId(),
+        base.getDescription()
+      )
     );
 
-    I18NString url = null;
-
     if (gtfsStop.getUrl() != null) {
-      url =
+      builder.withUrl(
         translationHelper.getTranslation(
           org.onebusaway.gtfs.model.Stop.class,
           "url",
           base.getId().getId(),
           gtfsStop.getUrl()
-        );
+        )
+      );
     }
 
-    return new Stop(
-      base.getId(),
-      name,
-      base.getCode(),
-      desc,
-      base.getCoordinate(),
-      base.getWheelchairAccessibility(),
-      base.getLevel(),
-      gtfsStop.getPlatformCode(),
-      fareZones,
-      url,
-      gtfsStop.getTimezone() == null ? null : TimeZone.getTimeZone(gtfsStop.getTimezone()),
-      TransitModeMapper.mapMode(gtfsStop.getVehicleType()),
-      null
-    );
-  }
+    if (gtfsStop.getZoneId() != null) {
+      builder.addFareZones(
+        FareZone.of(new FeedScopedId(gtfsStop.getId().getAgencyId(), gtfsStop.getZoneId())).build()
+      );
+    }
+
+    if (gtfsStop.getTimezone() != null) {
+      builder.withTimeZone(TimeZone.getTimeZone(gtfsStop.getTimezone()));
```

Comment:
Do we really want to use the very old `java.util.TimeZone`? I would strongly prefer to use `java.time.ZoneId` for new code.

Reply:
We will go over TimeZone/ZoneId when we make this configurable - fixed for the entire transit model as a separate task. So, I will leave this as is for now.